### PR TITLE
AddonManager: Render different addon documents on a tabbed group

### LINKED ISSUE #12968: [FEATURE] Add Changelog feature for addon devs to display what is new in an addon update
**Is your feature request related to a problem? Please describe.**
Not necessarily. It would be nice as an end user to know what was updated in the latest addon that needs to be updated. 

**Describe the solution you'd like**
Perhaps a dialog box or some sort of output when installing an update of a specific addon to output what has changed in this version update.

**Describe alternatives you've considered**
Providing a link back to the repository somehow?

**Additional context**
Several addon devs provide changelogs in their READMEs. For example:  
* https://github.com/mwganson/Editor_Assistant#changelog
* https://github.com/Zolko-123/FreeCAD_Assembly4#release-notes
* https://github.com/shaise/FreeCAD_SheetMetal#release-notes
* https://github.com/shaise/FreeCAD_FastenersWB#release-notes-berberic2

### PR DESCRIPTION
**Pull request by hasecilu**
Works as expected but probably it's better to tweak a little the code. Are plans to update the package metadata format? Not sure if it really interesting to h ave the license.

>[!IMPORTANT]
>As of now documents URL are hardcoded, probably is better to only hardcode the `package.xml` URL and then from it parse all needed URLS, but as of now contributing or changelog are not in the specification.

- [ ] Hide when showing macros
- [ ] Heading with link is downgraded to text
- [ ] Emojis from changelog don't appear :sob: 

image

Fix #12968

### CODE REVIEW

**Comment by luzpaz** [comment]
An update to the https://wiki.freecad.org/Package_Metadata (package.xml) documentation will be required as well.

**Comment by hasecilu** [comment]
I was making some tests like adding URLs on the metadata (https://github.com/instancezero/in3dca-freegrid/commit/63c04740820b90a9de8aa13b306aaa92429197da) but **would be really needed?** 

The URL of the README on the metadata is the `blob` one (html) which is not used by the Addon Manager to render, AM generates the raw URL containing markdown which is used. CONTRIBUTING and CHANGELOG are fairly standard files, license varies but at least we know the file.

**Is there another file for which would be useful to show to users?** I usually add a README file on the translations directory until now I only have put instructions there but maybe adding a table with progress for each language is informative.

**Comment by hasecilu** [comment]
I was thinking on extending the specification to contain the relative path of the document like:

```xml
<changelog>./CHANGELOG.md</changelog>
<contributing>CONTRIBUTING.md</contributing>
```

When we need raw or blob URL we can use the updated `def construct_git_url(repo, doc: str, raw_file: bool = True)` function that works with many git providers.


I don't like the README way because it directly assigns a branch on the URL:
```xml
  <url type="readme">https://github.com/instancezero/in3dca-freegrid/blob/main/README.md</url>
```

And when I add a custom URL for a testing branch, the AM still tries docs from main branch, but this is an edge case.

An alternative way would be to add entry like LICENSE. where file is specified 
```xml
  <license file="LICENSE">AGPL-3.0-or-later</license>
  <changelog file="CHANGELOG.md"></changelog>
  <contributing file="CONTRIBUTING.md"></contributing>
```

augmenting metadata spec with this entries we can construct the needed URLs.

File: src/Mod/AddonManager/Widgets/addonmanager_widget_package_details_view.py [hunk 1]
```diff
@@ -106,17 +109,29 @@ def __init__(self, parent: QtWidgets.QWidget = None):
...
         self.location_label = QtWidgets.QLabel(self)
         self.url_label = QtWidgets.QLabel(self)
         self.url_label.setOpenExternalLinks(True)
         self.location_label.setTextInteractionFlags(QtCore.Qt.TextSelectableByMouse)
+
+        self.tab_widget = QtWidgets.QTabWidget(self)
+        self.readme_browser = WidgetReadmeBrowser(self)
+        self.changelog_browser = WidgetReadmeBrowser(self)
+        self.contrib_browser = WidgetReadmeBrowser(self)
+        self.license_browser = WidgetReadmeBrowser(self)
```
Comment (hasecilu) [inline src/Mod/AddonManager/Widgets/addonmanager_widget_package_details_view.py:122]: Is LICENSE interesting enough for users or should remove it? Any other doc?

File: src/Mod/AddonManager/addonmanager_devmode.py [hunk 2]
```diff
@@ -249,6 +249,7 @@ def _populate_urls_from_metadata(self, metadata):
                 self.dialog.documentationURLLineEdit.setText(url["location"])
             elif url["type"] == "discussion":
                 self.dialog.discussionURLLineEdit.setText(url["location"])
+                # TODO: add changelog and contributing
```
Comment (hasecilu) [inline src/Mod/AddonManager/addonmanager_devmode.py:252]: If added to spec...

**Comment by ovo-Tim** [comment]
Hi, since you changed `construct_git_url`, maybe you could have a look at this:  #19242

File: src/Mod/AddonManager/Widgets/addonmanager_widget_package_details_view.py (hunk 1 above)
Comment (luzpaz) [reply src/Mod/AddonManager/Widgets/addonmanager_widget_package_details_view.py:122]: Yea, license makes sense.

File: src/Mod/AddonManager/Widgets/addonmanager_widget_package_details_view.py [hunk 3]
```diff
@@ -106,17 +109,29 @@ def __init__(self, parent: QtWidgets.QWidget = None):
...
+
+        # TODO: notify WB maintainers to include CHANGELOG and CONTRIBUTING files
```
Comment (luzpaz) [inline src/Mod/AddonManager/Widgets/addonmanager_widget_package_details_view.py:124]: We can open a tracking ticket on FreeCAD-Addons when this PR is merged.

File: src/Mod/AddonManager/addonmanager_devmode_metadata_checker.py [hunk 4]
```diff
@@ -141,6 +141,7 @@ def validate_urls(urls) -> List[str]:
         else:
             found_repo = False
             found_readme = False
+            # TODO: add changelo contributing
```
Comment (luzpaz) [inline src/Mod/AddonManager/addonmanager_devmode_metadata_checker.py:144]: 'changelog'

File: src/Mod/AddonManager/Widgets/addonmanager_widget_package_details_view.py (hunk 3 above)
Comment (hasecilu) [reply src/Mod/AddonManager/Widgets/addonmanager_widget_package_details_view.py:123]: Can the "Addon developers" group on on FreeCAD's GitHub org emit announcements?

File: src/Mod/AddonManager/addonmanager_metadata.py [hunk 5]
```diff
@@ -53,6 +54,23 @@ class License:
     file: str = ""
 
 
+class DocType(IntEnum):
+    changelog = 0
+    contributing = auto()
+    license = auto()
+    readme = auto()
+    picture = auto()  # TODO: for gallery
```
Comment (hasecilu) [inline src/Mod/AddonManager/addonmanager_metadata.py:62]: Could work like that if images are included on the repo, otherwise we could use URLs to the images.

https://github.com/FreeCAD/FreeCAD/issues/13632

File: src/Mod/AddonManager/addonmanager_metadata.py [hunk 6]
```diff
@@ -306,10 +333,14 @@ def _parse_child_element(namespace: str, child: ET.Element, metadata: Metadata):
...
             # List of licenses
-            metadata.license.append(MetadataReader._parse_license(child))
+            lic = MetadataReader._parse_license(child)
+            metadata.license.append(lic)
+            metadata.docs.append(Doc(doc=DocType.license, path=lic.file))
         elif tag == "url":
             # List of urls
             metadata.url.append(MetadataReader._parse_url(child))
+        elif tag == "doc":
+            metadata.docs.append(MetadataReader._parse_doc(child))
```
Comment (hasecilu) [inline src/Mod/AddonManager/addonmanager_metadata.py:343]: Created new tag
Tested adding testing branch to AM on settings -https://github.com/instancezero/in3dca-freegrid/tree/test/am_docs

```xml
  <doc type="readme">README.md</doc>
  <doc type="changelog">CHANGELOG.md</doc>
  <doc type="contributing">CONTRIBUTING.md</doc>
```
